lib/services/internalEngine/index.js

```diff
@@ -300,153 +263,145 @@ class InternalEngine {
+        if (! exists) {
+          errorsManager.throw('document_not_found', id);
+        }
 
-    // extends the response with the source from request
-    // When we write in ES, the response from it doesn't contain the initial document content
-    return this.client.index(request)
-      .then(result => _.extend(result, {_source: content}))
-      .catch(error => Bluebird.reject(this.esWrapper.formatESError(error)));
+        return this.createOrReplace(collection, id, body, options);
+      });
   }
 
   /**
-   * Performs a partial update to a document
+   * Send to elasticsearch the new document
    *
-   * @param {string} type - data collection
-   * @param {string} id of the document to update
-   * @param {object} updateContent
+   * @param {string} collection - collection name
+   * @param {string} id - document ID
+   * @param {object} body
    * @param {object} [options]
-   * @returns {Promise} resolve an object that contains _id
+   * @returns {Promise}
    */
-  update (type, id, updateContent, options = {}) {
-    const request = {
-      type,
-      id,
-      index: this.index,
-      body: {
-        doc: updateContent
-      }
-    };
+  create (collection, id, body, options = {}) {
+    const promise = id
+      ? this.exists(collection, id)
+      : Promise.resolve({ body: false });
 
-    if (options.refresh === 'wait_for') {
-      request.refresh = 'wait_for';
-    }
-    if (options.retryOnConflict !== undefined) {
-      request.retryOnConflict = options.retryOnConflict;
-    }
-    else {
-      request.retryOnConflict = this.config.defaults.onUpdateConflictRetries;
-    }
+    return promise
+      .then(({ body: exists }) => {
+        if (exists) {
+          errorsManager.throw('document_already_exists', id);
+        }
 
-    return this.client.update(request)
-      .catch(error => Bluebird.reject(this.esWrapper.formatESError(error)));
+        return this.createOrReplace(collection, id, body, options);
+      });
   }
 
   /**
-   * Replace a document with new content
+   * Performs a partial update to a document
    *
-   * @param {string} type - data collection
-   * @param {string} id - document ID
-   * @param {object} content
+   * @param {string} collection - collection name
+   * @param {string} id of the document to update
+   * @param {object} updateContent
    * @param {object} [options]
-   * @returns {Promise}
+   * @returns {Promise} resolve an object that contains _id
    */
-  replace (type, id, content, options = {}) {
-    // extends the response with the source from request
-    // When we write in ES, the response from it doesn't contain the initial document content
-    return this.exists(type, id)
-      .then(exists => {
+  update (collection, id, updateContent, options = {}) {
+    return this.exists(collection, id)
+      .then(({ body: exists }) => {
+        if (! exists) {
+          errorsManager.throw('document_not_found', id);
+        }
+
         const request = {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Group all shorthand properties at either the beginning or end of this object declaration. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3499)